stingray/tests/test_base.py

```diff
@@ -47,43 +49,222 @@ def __init__(self, dummy=None):
-            assert not hasattr(new_so, attr)
-            continue
-        so_attr = getattr(so, attr)
-        new_so_attr = getattr(new_so, attr)
-        assert so_attr == new_so_attr
-
-
 class TestStingrayObject:
     @classmethod
     def setup_class(cls):
-        cls.arr = [4, 5, 2]
+        cls.arr = np.asarray([4, 5, 2])
         sting_obj = DummyStingrayObj(cls.arr)
         sting_obj.pardulas = [3.0 + 1.0j, 2.0j, 1.0 + 0.0j]
+        sting_obj.sebadas = [[0, 1], [2, 3], [4, 5]]
+        sting_obj._sebadas = [[0, 1], [2, 3], [4, 5]]
         sting_obj.pirichitus = 4
         sting_obj.parafritus = "bonus!"
         sting_obj.panesapa = [[41, 25], [98, 3]]
         cls.sting_obj = sting_obj
 
+    def test_print(self):
+        print(self.sting_obj)
+
     def test_preliminary(self):
         assert np.allclose(self.sting_obj.guefus, self.arr)
 
     def test_instantiate_without_main_array_attr(self):
         with pytest.raises(RuntimeError):
             BadStingrayObj(self.arr)
 
+    def test_equality(self):
+        ts1 = copy.deepcopy(self.sting_obj)
+        ts2 = copy.deepcopy(self.sting_obj)
+        assert ts1 == ts2
+
+    def test_different_array_attributes(self):
+        ts1 = copy.deepcopy(self.sting_obj)
+        ts2 = copy.deepcopy(self.sting_obj)
+        # Add a meta attribute only to ts1. This will fail
+        ts1.blah = 2
+        assert ts1 != ts2
+
+        # Add a non-scalar meta attribute, but with the same name, to ts2.
+        ts2.blah = [2]
+        assert ts1 != ts2
+
+        # Get back to normal
+        del ts1.blah, ts2.blah
+        assert ts1 == ts2
+
+        # Add a non-scalar meta attribute to both, just slightly different
+        ts1.blah = [2]
+        ts2.blah = [3]
+        ts1 != ts2
+
+        # Get back to normal
+        del ts1.blah, ts2.blah
+        assert ts1 == ts2
+
+        # Add a meta attribute only to ts2. This will also fail
+        ts2.blah = 3
+        assert ts1 != ts2
+
+    def test_different_meta_attributes(self):
+        ts1 = copy.deepcopy(self.sting_obj)
+        ts2 = copy.deepcopy(self.sting_obj)
+        # Add an array attribute to ts1. This will fail
+        ts1.blah = ts1.guefus
+        assert ts1 != ts2
+
+        # Get back to normal
+        del ts1.blah
+        assert ts1 == ts2
+        # Add an array attribute to ts2. This will fail
+        ts2.blah = ts1.guefus
+        assert ts1 != ts2
+
+        # Get back to normal
+        del ts2.blah
+        assert ts1 == ts2
+
+    @pytest.mark.parametrize("inplace", [True, False])
+    def test_apply_mask(self, inplace):
+        ts = copy.deepcopy(self.sting_obj)
+        obj = ts.apply_mask([True, True, False], inplace=inplace)
+
+        assert obj.parafritus == "bonus!"
+        assert np.array_equal(obj.guefus, [4, 5])
+        assert np.array_equal(obj.panesapa, ts.panesapa)
+        assert np.array_equal(obj.pardulas, [3.0 + 1.0j, 2.0j])
+        assert np.array_equal(obj.sebadas, [[0, 1], [2, 3]])
```

Comment:
For my own entertainment, what are all those words, which I'm assuming are Italian? :D

Reply:
They are all Sardinian desserts, in Sardinian language:
https://www.italiadelight.it/discover-your-favourite-traditional-sardinian-dessert/?lang=en